src/v/iceberg/datatypes_json.cc

```diff
@@ -177,6 +177,10 @@ void rjson_serialize(iceberg::json_writer& w, const iceberg::nested_field& f) {
     w.Bool(bool(f.required));
     w.Key("type");
     rjson_serialize(w, f.type);
+    w.Key("initial-default");
+    w.Null();
+    w.Key("write-default");
+    w.Null();
```

Comment:
Wait we make everything optional? Which is why this is valid? Should we only include these if `f.required`?

Reply:
Oh I see it's a later commit.... Should we reorder? :)

Reply:
yeah fair point.

Reply:
done